sierra_adapter/sierra_merger/src/main/scala/uk/ac/wellcome/platform/sierra_merger/services/SierraMergerWorkerService.scala

```diff
@@ -13,17 +14,25 @@ import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.util.{Failure, Success}
 
-class SierraBibMergerWorkerService @Inject()(
+class SierraMergerWorkerService @Inject()(
   reader: SQSReader,
   system: ActorSystem,
   metrics: MetricsSender,
-  sierraBibMergerUpdaterService: SierraBibMergerUpdaterService
+  sierraBibMergerUpdaterService: SierraMergerUpdaterService,
+  @Flag("sierra.resourceType") resourceType: String
 ) extends SQSWorker(reader, system, metrics)
     with Logging {
 
   override def processMessage(message: SQSMessage): Future[Unit] =
-    JsonUtil.fromJson[SierraBibRecord](message.body) match {
-      case Success(record) => sierraBibMergerUpdaterService.update(record)
-      case Failure(e) => Future(logger.warn(s"Failed processing $message", e))
+    resourceType match {
+      case "bibs" => JsonUtil.fromJson[SierraBibRecord](message.body) match {
```

Comment:
So I’d add a similar case for "items" just below…

Reply:
I don't think I like this: I'd go for something similar to the transformer where we read the flag once in `TransformableParserModule` and based on that flag we instantiate the right `TransformableParser` and from then onwards use that. At runtime I don't think we should be using this flag